corehq/apps/app_manager/dbaccessors.py

```diff
@@ -500,16 +501,24 @@ def get_version_build_id(domain, app_id, version):
 def get_case_types_from_apps(domain):
     """
     Get the case types of modules in applications in the domain.
+    Also returns case types for SaveToCase properties in the domain, if the toggle is enabled.
     :returns: A set of case_types
     """
+    save_to_case_updates = set()
+    if VELLUM_SAVE_TO_CASE.enabled(domain):
+        for app in get_apps_in_domain(domain):
+            for form in app.get_forms():
+                for update in form.get_save_to_case_updates():
+                    save_to_case_updates.add(update)
+
     case_types_agg = NestedAggregation('modules', 'modules').aggregation(
         TermsAggregation('case_types', 'modules.case_type.exact'))
     q = (AppES()
          .domain(domain)
          .is_build(False)
          .size(0)
          .aggregation(case_types_agg))
-    return set(q.run().aggregations.modules.case_types.keys) - {''}
+    return set(q.run().aggregations.modules.case_types.keys).union(save_to_case_updates) - {''}
```

Comment:
Then I'd change this line into two lines:
```python
case_types = set(q.run().aggregations.modules.case_types.keys)
return save_to_case_case_types.union(case_types) - {''}
```

Reply:
Some good points, the code readability could definitely be improved. I have done some changes as per your suggestions. Thanks for pointing them out :+1: